File: examples/leetcode/1480-running-sum-1d-array/1480.go
package main
import "fmt"

func runningSumOld(nums []int) []int {
	len  := len(nums)
	sum  := 0 
	rSum := make([] int, len)

	for k, v := range nums {
		sum += v
		rSum[k] = sum
	}

	return rSum
}

func runningSum(nums []int) []int {
	numLen  := len(nums)
	runSum  := make([]int, numLen)
	currSum := 0

	for k, v := range nums {
		currSum += v
		runSum [k] = currSum
	}

	return runSum
}

func main() {
	fmt.Println(runningSum([]int{1, 2, 3}))
}

File: examples/leetcode/1480-running-sum-1d-array/1480.php
<?php

function runningSum($nums) {
    $runningSum = [];
    $len = count($nums);
    $sum = 0;
    for ($i = 0; $i < $len; $i++) {
        $sum += $nums [$i];
        $runningSum[] = $sum;
    }

    return $runningSum;
}

print_r(runningSum([1])) . "\n";
print_r(runningSum([1, 2])) . "\n";
print_r(runningSum([1, 2, 3])) . "\n";

print_r(runningSum([0, 0, 0])) . "\n";